src/vs/base/node/powershell.ts

```diff
@@ -0,0 +1,326 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as pfs from 'vs/base/node/pfs';
+import * as os from 'os';
+import * as path from 'vs/base/common/path';
+import { env } from 'vs/base/common/process';
+
+const WindowsPowerShell64BitLabel = 'Windows PowerShell';
+const WindowsPowerShell32BitLabel = 'Windows PowerShell (x86)';
+
+// This is required, since parseInt("7-preview") will return 7.
+const IntRegex: RegExp = /^\d+$/;
+
+const PwshMsixRegex: RegExp = /^Microsoft.PowerShell_.*/;
+const PwshPreviewMsixRegex: RegExp = /^Microsoft.PowerShellPreview_.*/;
+
+// The platform details descriptor for the platform we're on
+const isProcess64Bit: boolean = process.arch === 'x64';
+const isOS64Bit: boolean = isProcess64Bit || env.hasOwnProperty('PROCESSOR_ARCHITEW6432');
+
+export interface IPowerShellExeDetails {
+	readonly displayName: string;
+	readonly exePath: string;
+}
+
+export interface IPossiblePowerShellExe extends IPowerShellExeDetails {
+	exists(): boolean;
+}
+
+class PossiblePowerShellExe implements IPossiblePowerShellExe {
+	public readonly exePath: string;
+	public readonly displayName: string;
+
+	private knownToExist: boolean | undefined;
+
+	constructor(
+		pathToExe: string,
+		installationName: string,
+		{ knownToExist = false }: { knownToExist?: boolean } = {}) {
+
+		this.exePath = pathToExe;
+		this.displayName = installationName;
+		this.knownToExist = knownToExist || undefined;
+	}
+
+	public exists(): boolean {
+		if (this.knownToExist === undefined) {
+			this.knownToExist = pfs.fileExistsSync(this.exePath);
+		}
+		return this.knownToExist;
+	}
+}
+
+function getProgramFilesPath(
+	{ useAlternateBitness = false }: { useAlternateBitness?: boolean } = {}): string | null {
+
+	if (!useAlternateBitness) {
+		// Just use the native system bitness
+		return env.ProgramFiles!;
+	}
+
+	// We might be a 64-bit process looking for 32-bit program files
+	if (isProcess64Bit) {
+		return env['ProgramFiles(x86)']!;
+	}
+
+	// We might be a 32-bit process looking for 64-bit program files
+	if (isOS64Bit) {
+		return env.ProgramW6432!;
+	}
+
+	// We're a 32-bit process on 32-bit Windows, there is no other Program Files dir
+	return null;
+}
+
+function getSystem32Path({ useAlternateBitness = false }: { useAlternateBitness?: boolean } = {}): string {
+	const windir: string = env.windir!;
+
+	if (!useAlternateBitness) {
+		// Just use the native system bitness
+		return path.join(windir, 'System32');
+	}
+
+	// We might be a 64-bit process looking for 32-bit system32
+	if (isProcess64Bit) {
+		return path.join(windir, 'SysWOW64');
+	}
```

Comment:
Would you rather not have the option of running the other bitness because of those issues? If running in x64 vscode, you only get x64 PowerShell, etc.
(This can't happen with Windows ARM because that uses x86 Windows PowerShell)

Ideally that bug should be fixed in conpty/winpty but I know that's easier said than done...

Reply:
I can't remember the combination that doesn't work but it's probably best not to route people to things that will break. Not a big issue though.